src/vuex/modules/connection.js

```diff
@@ -17,11 +20,27 @@ export default function() {
   }
 
   const actions = {
-    async checkForPersistedNetwork({ commit }) {
-      const network = localStorage.getItem(`network`)
-      if (network) {
-        await commit(`setNetworkId`, JSON.parse(network))
-      }
+    async checkForPersistedNetwork({ dispatch, commit }) {
+      const network = JSON.parse(localStorage.getItem(`network`))
+      dispatch(`checkAvailableNetworks`).then( async result => {
+        let availNetworks = result.data.data.networks
+        let availNetworksID = Object.values(availNetworks).map(
+          network => network.id
+        )
+        // Little hack to reorder the array so 'cosmos-hub-mainnet' stays at the beginning
```

Comment:
Instead of a complicated filtering, how about just hardcoding the default network for now?